pkg/controller/wip/controller_test.go

```diff
@@ -138,6 +165,125 @@ func TestReconcileBroker(t *testing.T) {
 
 }
 
+func TestReconcileBrokerErrorFetchingCatalog(t *testing.T) {
+	fakeKubeClient, fakeCatalogClient, fakeBrokerCatalog, _, _, testController, _, stopCh := newTestController(t)
+	defer close(stopCh)
+
+	fakeBrokerCatalog.RetErr = fakebrokerapi.ErrInstanceNotFound
+	broker := getTestBroker()
+
+	testController.reconcileBroker(broker)
+
+	actions := filterActions(fakeCatalogClient.Actions())
+	if e, a := 1, len(actions); e != a {
+		t.Fatalf("Unexpected number of actions: expected %v, got %v. Actions: %+v", e, a, actions)
+	}
+	updateAction := actions[0].(core.UpdateAction)
+	if e, a := "update", updateAction.GetVerb(); e != a {
+		t.Fatalf("Unexpected verb on actions[1]; expected %v, got %v", e, a)
+	}
+	updateObject := updateAction.GetObject().(*v1alpha1.Broker)
+	if e, a := broker.Name, updateObject.Name; e != a {
+		t.Fatalf("Unexpected name of broker: expected %v, got %v", e, a)
+	}
+	if e, a := v1alpha1.ConditionFalse, updateObject.Status.Conditions[0].Status; e != a {
+		t.Fatalf("Unexpected condition status: expected %v, got %v", e, a)
+	}
+
+	kubeActions := fakeKubeClient.Actions()
+	if e, a := 0, len(kubeActions); e != a {
+		t.Fatalf("Unexpected number of actions: expected %v, got %v", e, a)
+	}
+}
+
+func TestReconcileBrokerWithAuthError(t *testing.T) {
+	fakeKubeClient, fakeCatalogClient, _, _, _, testController, _, stopCh := newTestController(t)
+	defer close(stopCh)
+
+	broker := getTestBroker()
+	broker.Spec.AuthSecret = &v1.ObjectReference{
```

Comment:
I think it would be good to add a test where there is no secret at all specified -meaning the broker doesn't require any auth. I don't think our controller should flag that as an error - if the broker does require an id/pwd then things should fail during the GET /v2/catalog phase.  As of now, from looking at the code, I think we're ok but we should add a test to make sure we don't break current functionality by mistake. Follow-on PR is ok if you want.